docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1145 @@
+:branch:          1.5
+:ls_version:		  1.5.0.beta1
+
+= How to write a Logstash {plugintype} plugin
+
+To develop a new {plugintype} for Logstash, you build a self-contained Ruby gem
+that lives in its own GitHub repository. You can use the example {plugintype}
+implementation as a starting point. (If you're unfamiliar with Ruby, you can
+find an excellent quickstart guide at
+https://www.ruby-lang.org/en/documentation/quickstart/[].)
+
+NOTE: As of Logstash 1.5, all plugins are self-contained Ruby gems. This change
+makes it possible to develop and release plugins separately. In previous
+versions, plugins were part of the core Logstash distribution.
+
+== Get started
+
+{getstarted}
+
+=== Create a GitHub repo for the new plugin
+Each Logstash plugin lives in its own GitHub repository. To create a new repository for your plugin:
+
+. Log in to GitHub.
+. Click the **Repositories** tab. You'll see a list of other repositories you've forked or contributed to.
+. Click the green **New** button in the upper right.
+. Specify the following settings for your new repo:
+** **Repository name** -- a unique name of the form logstash-{plugintype}-pluginname.
+** **Public or Private** -- your choice, but the repository must be Public if you want to submit it as an official plugin.
+** **Initialize this repository with a README** -- enables you to immediately clone the repository to your computer.
+. Click **Create Repository**.
+
+=== Copy the {plugintype} code
+
+==== Build your local repository
+. **Clone your plugin.** Replace `GITUSERNAME` with your github username, and
+`MYPLUGINNAME` with your plugin name.
+** `git clone https://github.com/GITUSERNAME/logstash-{plugintype}-MYPLUGINNAME.git`
+*** alternately, via ssh: `git clone git@github.com:GITUSERNAME/logstash-{plugintype}-MYPLUGINNAME.git`
+** `cd logstash-{plugintype}-MYPLUGINNAME`
+
+. **Clone the {inputtype} plugin example and copy it to your plugin branch.**
++
+You don't want to include the example .git directory or its contents, so delete
+it before you copy the example.
++
+** `cd /tmp`
+** `git clone https://github.com/logstash-plugins/logstash-{plugintype}-{pluginname}.git`
+** `cd logstash-{plugintype}-{pluginname}`
+** `rm -rf .git`
+** `cp -R * /path/to/logstash-{plugintype}-mypluginname/`
```

Comment:
For ultimate copy and pastiness, we could include

`mkdir -p /path/to/logstash-{plugintype}-mypluginname/`